mailbox/api/src/main/java/org/apache/james/mailbox/indexer/ReIndexer.java

```diff
@@ -19,14 +19,20 @@
 
 package org.apache.james.mailbox.indexer;
 
+import org.apache.james.core.User;
+import org.apache.james.mailbox.MessageUid;
 import org.apache.james.mailbox.exception.MailboxException;
 import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.task.Task;
 
 public interface ReIndexer {
 
+    Task reIndex(User user) throws MailboxException;
+
     Task reIndex(MailboxPath path) throws MailboxException;
 
+    Task reIndex(MailboxPath path, MessageUid uid) throws MailboxException;
```

Comment:
what about also providing reIndex(MessageId id)?

Reply:
Added in yet another PR https://github.com/linagora/james-project/pull/1808